solana/migration/src/api/add_liquidity.rs

```diff
@@ -97,11 +97,11 @@ pub fn add_liquidity(
     let share_amount = if accs.from_mint.decimals > accs.to_mint.decimals {
         data.amount
             .checked_mul(10u64.pow((accs.from_mint.decimals - accs.to_mint.decimals) as u32))
-            .unwrap()
+            .ok_or_else(|| Result::Err(SolitaireError::InsufficientFunds))?
```

Comment:
The return type for `ok_or_else` should be just the error type, not `Result<T, E>`.  Also there's no need for lazy evaluation here so we can use `ok_or` instead:

```suggestion
            .ok_or(SolitaireError::InsufficientFunds)?
```